packages/smart-wallet/src/smartWallet.js

```diff
@@ -1068,14 +1079,21 @@ export const prepareSmartWallet = (baggage, shared) => {
           const { myStore } = this.state;
           const { resultStepWatcher } = this.facets;
 
-          const { targetName: name, method, args, saveResult } = message;
+          const { targetName: name, method, args, saveResult, id } = message;
           myStore.has(name) || Fail`cannot invoke ${q(name)}: no such item`;
           const value = myStore.get(name);
           trace('entry', name, value);
           trace('invoke', value, '.', method, '(', args, ')');
+          if (id) {
```

Comment:
Isn't `id` mandatory ? Or are we expecting to only update/watch if non-falsy? I'd prefer making `id` optional to make that clearer.

Reply:
The type needs updating then